### PR TITLE
adding distributed loki configs and updates to scripts

### PR DESCRIPTION
**Pull request by memodi**
- updating Jenkinsfile for awscreds and netobserv.sh
- adding lokiStack configs
- adding script to deploy AWS s3 secret
- updating Readme.
- deploying NOO from main instead of from git repo

### CODE REVIEW

File: scripts/netobserv-catalogsource.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: netobserv-testing
+  namespace: openshift-marketplace
+spec:
+  displayName: NetObservTest
+  image: 'quay.io/rhn_support_memodi/network-observability-operator-catalog:vmain'
```
Comment (memodi) [inline scripts/netobserv-catalogsource.yaml:8]: I will update this image location to netobserv org repo in quay once I have other PR for git workflows up.

File: Jenkinsfile [hunk 2]
```diff
@@ -178,20 +177,42 @@ pipeline {
...
+                    withCredentials([file(credentialsId: 'b73d6ed3-99ff-4e06-b2d8-64eaaf69d1db', variable: 'OCP_AWS')]) {
+                        script {
+                            returnCode = sh(returnStatus: true, script: """
+                                mkdir -p ~/.aws
+                                cp -f $OCP_AWS ~/.aws/credentials
+                                echo "[profile default]
+                                region = `cat $WORKSPACE/flexy-artifacts/workdir/install-dir/terraform.platform.auto.tfvars.json | jq -r ".aws_region"`
+                                output = text" > ~/.aws/config
+                            """)
+
```
Comment (nathan-weinberg) [inline Jenkinsfile:191]: Remove empty line here please

File: scripts/README.md [hunk 3]
```diff
@@ -95,3 +91,21 @@ To see all command line options available for the NOPE tool, you can run it with
 ## Fetching metrics using touchstone 
 NetObserv metrics uploaded to elasticsearch can be fetched using `touchstone` tool provided by [benchmark-comparison](https://github.com/cloud-bulldozer/benchmark-comparison). Once you have touchstone setup, you can run command as:
 `$ touchstone_compare/bin/touchstone_compare --database elasticsearch -url <elasticsearch instance:port> -u <run uuid> --config=scripts/netobserv_touchstone.json`
+
+## Creating LokiStack using Loki Operator
```
Comment (nathan-weinberg) [inline scripts/README.md:95]: Can we put this section in chronological order of when it should be done in the setup?

File: scripts/deploy-loki-aws-secret.sh [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      exit 1
+  fi
+}
+
+#Create s3 bucket 
+create_s3_bucket(){
+  if aws configure get region; then
+      echo "bucket: Create s3 bucket $LOKI_BUCKET_NAME if not exist"
+  else
+      echo "bucket: Warning: Can not connect to aws, Insure bucket ready before deploy lokistack"
```
Comment (nathan-weinberg) [inline scripts/deploy-loki-aws-secret.sh:54]:
```suggestion
      echo "bucket: Warning: Can not connect to AWS, ensure bucket is ready before trying to deploy lokistack"
```

File: scripts/deploy-loki-aws-secret.sh [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  if [[ $AWS_ACCESS_KEY_ID == "" || ${AWS_SECRET_ACCESS_KEY} == "" ]]; then
+      echo "CreateAWSSecret: Error: option no value! AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID,AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY"
+      exit 1
+  fi
+}
+
+#Create s3 bucket 
+create_s3_bucket(){
+  if aws configure get region; then
+      echo "bucket: Create s3 bucket $LOKI_BUCKET_NAME if not exist"
```
Comment (nathan-weinberg) [inline scripts/deploy-loki-aws-secret.sh:52]: Can't tell from this message what is going on here

File: scripts/deploy-loki-aws-secret.sh [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  if [[ $? == 0 ]]  ;then
+     echo "bucket: use existing bucket"
+  else
+      aws s3api create-bucket --bucket $LOKI_BUCKET_NAME --region $AWS_DEFAULT_REGION --create-bucket-configuration LocationConstraint=$AWS_DEFAULT_REGION
+      if [[ $? == 0 ]]  ;then
+         echo "bucket: created new bucket $LOKI_BUCKET_NAME"
+      else
+	 echo "bucket: Error: fail to create bucket $LOKI_BUCKET_NAME!"
+	 exit 1
+      fi
```
Comment (nathan-weinberg) [inline scripts/deploy-loki-aws-secret.sh:69]: Looks like an indentation error here

**Review by nathan-weinberg — COMMENTED** [review]
Can you link a successful Jenkins job run @memodi?

File: scripts/deploy-loki-aws-secret.sh (hunk 5 above)
Comment (memodi) [reply scripts/deploy-loki-aws-secret.sh:52]: I just grabbed this script from [here](https://gitlab.cee.redhat.com/aosqe/aosqe-tools/-/blob/master/logging/log_template/loki-operator/deploy_s3-secret.sh) and didn't make any modifications except to change some default options. I'll resubmit with formatting and fixing typos.

**Comment by memodi** [comment]
Adding Success runs:
Deployed through [OperatorHub](https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/memodi-e2e-benchmarking-multibranch-pipeline/job/distributed-loki/35/)
Deployed  from [Source](https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/memodi-e2e-benchmarking-multibranch-pipeline/job/distributed-loki/35/)
Link to [re-arranged Dashboard](http://dittybopper-dittybopper.apps.memodi-08231012.qe.devcluster.openshift.com/d/hmI9De_nz/netobserv-metrics?orgId=1)

File: Jenkinsfile [hunk 7]
```diff
@@ -79,6 +79,16 @@ pipeline {
                 If None is selected the installation will be skipped
             '''
         )
+        choice(
+            name: 'LokiStack_Size',
+            choices: ['1x.extra-small', '1x.small', '1x.medium'],
+            description: '''
+                Depending on size of cluster nodes, use following guidance to choose LokiStack size:<br/>
+                1x.extra-small -  Nodes size < m5.4xlarge<br/>
```
Comment (nathan-weinberg) [inline Jenkinsfile:87]:
```suggestion
                1x.extra-small - Nodes size < m5.4xlarge<br/>
```

File: Jenkinsfile [hunk 8]
```diff
@@ -79,6 +79,16 @@ pipeline {
                 If None is selected the installation will be skipped
             '''
         )
+        choice(
+            name: 'LokiStack_Size',
+            choices: ['1x.extra-small', '1x.small', '1x.medium'],
```
Comment (nathan-weinberg) [inline Jenkinsfile:84]: Note that first choice is default - assuming most Flexy clusters have node sizes smaller than m5.4xlarge?

File: scripts/README.md [hunk 9]
```diff
@@ -30,7 +30,23 @@ kube:admin
 There are two methods you can use to install the operator:
 - To install from Operator Hub, navigate to the `scripts/` directory and run `$ INSTALLATION_SOURCE=OperatorHub; source netobserv.sh ; deploy_netobserv`
 - To install from Source, navigate to the `scripts/` directory and run `$ INSTALLATION_SOURCE=Source; source netobserv.sh ; deploy_main_catalogsource; deploy_netobserv`
+### Creating LokiStack using Loki Operator
```
Comment (nathan-weinberg) [inline scripts/README.md:33]:
```suggestion
- To install from Source, navigate to the `scripts/` directory and run `$ INSTALLATION_SOURCE=Source; source netobserv.sh ; deploy_main_catalogsource; deploy_netobserv`

### Creating LokiStack using Loki Operator
```

File: scripts/README.md [hunk 10]
```diff
@@ -30,7 +30,23 @@ kube:admin
...
+    * lokistack-1x-small.yaml - Small t-shirt size LokiStack
+        - Requirements: `m5.4xlarge` machines.
+        - Use case: Scale/Performance testing.
+    * lokistack-1x-medium.yaml - Medium t-shirt size LokiStack
+        - Requirments: `m5.8xlarge` machines.
+        - Use case: Large scale performance testing.
 
+    Depending upon your cluster size and use case, run `$ oc apply -f <lokistack yaml manifest>`
+4. LokiStack should be created under `openshift-operators-redhat` NS
 ### Setting up FLP service and creating service-monitor
```
Comment (nathan-weinberg) [inline scripts/README.md:50]:
```suggestion
4. LokiStack should be created under `openshift-operators-redhat` NS

### Setting up FLP service and creating service-monitor
```

**Review by nathan-weinberg — COMMENTED** [review]
Few additional minor comments but overall it is looking good to me

File: Jenkinsfile [hunk 11]
```diff
@@ -80,6 +79,16 @@ pipeline {
                 If None is selected the installation will be skipped
             '''
         )
+        choice(
+            name: 'LokiStack_Size',
```
Comment (nathan-weinberg) [inline Jenkinsfile:83]: Let's make this `LOKISTACK_SIZE` to match the other variables

File: Jenkinsfile (hunk 8 above)
Comment (memodi) [reply Jenkinsfile:84]: Yeah, mostly default to day to day use case. Also have tried to recommend based on use cases [here](https://github.com/openshift-qe/ocp-qe-perfscale-ci/pull/236/files#diff-25b96aa16f8b9f68c692d0db061713707e9cfc67b296e0ea30908928fb3ffcd6R40) . For next set of Perf runs we should run with m5.4xlarge machines.

**Pull request merged**